pkg/reconciler/ingress/ingress.go

```diff
@@ -148,83 +148,84 @@ func (c *Reconciler) reconcileIngress(ctx context.Context, ing *v1alpha1.Ingress
 
 	// TODO: check Gateway readiness before reporting Ingress ready
 	if routesReady {
-		var publicLbs, privateLbs []v1alpha1.LoadBalancerIngressStatus
+		lbs, err := c.lookUpLoadBalancers(ing, pluginConfig)
 
-		externalGateway := pluginConfig.ExternalGateway()
-		localGateway := pluginConfig.LocalGateway()
-
-		publicLbs, err = c.determineLoadBalancerIngressStatus(externalGateway)
-		if err != nil {
-			if apierrs.IsNotFound(err) {
-				ing.Status.MarkLoadBalancerFailed(
-					"GatewayDoesNotExist",
-					fmt.Sprintf(
-						"could not find Gateway %s/%s",
-						externalGateway.Namespace,
-						externalGateway.Name,
-					),
-				)
-				return fmt.Errorf("Gateway %s does not exist: %w", externalGateway.Name, err) //nolint:stylecheck
-			}
-			ing.Status.MarkLoadBalancerNotReady()
-			return err
-		}
-
-		privateLbs, err = c.determineLoadBalancerIngressStatus(localGateway)
 		if err != nil {
-			if apierrs.IsNotFound(err) {
-				ing.Status.MarkLoadBalancerFailed(
-					"GatewayDoesNotExist",
-					fmt.Sprintf(
-						"could not find Gateway %s/%s",
-						localGateway.Namespace,
-						localGateway.Name,
-					),
-				)
-				return fmt.Errorf("Gateway %s does not exist: %w", localGateway.Name, err) //nolint:stylecheck
-			}
 			ing.Status.MarkLoadBalancerNotReady()
 			return err
 		}
-
-		ing.Status.MarkLoadBalancerReady(publicLbs, privateLbs)
+		ing.Status.MarkLoadBalancerReady(lbs[v1alpha1.IngressVisibilityExternalIP], lbs[v1alpha1.IngressVisibilityClusterLocal])
 	} else {
 		ing.Status.MarkLoadBalancerNotReady()
 	}
 
 	return nil
 }
 
-// determineLoadBalancerIngressStatus will return the address for the Gateway.
-// If a service is provided, it will return the address of the service.
-// Otherwise, it will return the first address in the Gateway status.
-func (c *Reconciler) determineLoadBalancerIngressStatus(gwc config.Gateway) ([]v1alpha1.LoadBalancerIngressStatus, error) {
-	if gwc.Service != nil {
-		return []v1alpha1.LoadBalancerIngressStatus{
-			{DomainInternal: network.GetServiceHostname(gwc.Service.Name, gwc.Service.Namespace)},
-		}, nil
+// lookUpLoadBalancers will return a map of visibilites to
+// LoadBalancerIngressStatuses for the current Gateways in use.
+func (c *Reconciler) lookUpLoadBalancers(ing *v1alpha1.Ingress, gpc *config.GatewayPlugin) (map[v1alpha1.IngressVisibility][]v1alpha1.LoadBalancerIngressStatus, error) {
+	externalStatuses, err := c.collectLBIngressStatus(ing, gpc.ExternalGateways)
```

Comment:
nit: I created a helper method to just to return the single Gateway `gpc.ExternalyGateway()`